src/common/SurgeMessage.h

```diff
@@ -0,0 +1,55 @@
+/* 
+**  SurgeMessage provides basic functions for the synth to report messages back to the user
+**  with a platform-neutral API. A platform-specific version of each implementation is
+**  registered at startup.
+**
+**  Absent a gui, there is a simple stderr based implementation. In the initializer of SurgeGUIEditor
+**  we register GUI-based versions also.
+**
+**  You can call this with
+**
+**    SurgeMessage::reportProblem()
+**
+**  or what not. SurgeMessage is guaranteed not to be null once statics are initialized.
+*/
+
+#pragma once
+
+#include <string>
+#include <atomic>
+#include "SurgeError.h"
+
+class SurgeMessage
```

Comment:
The class name should be `ErrorMessage` (here also lets start putting stuff to Surge namespace instead of prefixing).

Reply:
Obviously, the class name needs to be changed to something else. `Surge::Error` is the message object and this is something else.

Reply:
Maybe it could be `ErrorReporter`. Nothing better comes to mind.